apps/storybook/src/TiledHeatmap.stories.tsx

```diff
@@ -0,0 +1,208 @@
+          break;
+        }
+      }
+    }
+  }
+  return array;
+}
+
+interface TileKey {
+  lod: number;
+  offset: Vector2;
+}
+
+class MandelbrotTilesApi extends TilesApi {
+  public readonly xDomain: Domain;
+  public readonly yDomain: Domain;
+  private readonly store;
+
+  public constructor(
+    size: Size,
+    tileSize: Size,
+    xDomain: Domain,
+    yDomain: Domain
+  ) {
+    super(tileSize, getLevelsSize(size, tileSize));
+    this.xDomain = xDomain;
+    this.yDomain = yDomain;
+
+    this.store = createFetchStore(
+      async (tile: TileKey) => {
+        const { lod, offset } = tile;
+        const levelSize = this.levelsSize[lod];
+        // Clip slice to size of the level
+        const width = clamp(levelSize.width - offset.x, 0, this.tileSize.width);
+        const height = clamp(
+          levelSize.height - offset.y,
+          0,
+          this.tileSize.height
+        );
+
+        const xScale = (this.xDomain[1] - this.xDomain[0]) / levelSize.width;
+        const xRange: Domain = [
+          this.xDomain[0] + xScale * offset.x,
+          this.xDomain[0] + xScale * (offset.x + width),
+        ];
+        const yScale = (this.yDomain[1] - this.yDomain[0]) / levelSize.height;
+        const yRange: Domain = [
+          this.yDomain[0] + yScale * offset.y,
+          this.yDomain[0] + yScale * (offset.y + height),
+        ];
+
+        return mandelbrot(50, xRange, yRange, { width, height });
+      },
+      {
+        type: 'Map',
+        areEqual: (a: TileKey, b: TileKey) =>
+          a.lod === b.lod && a.offset.equals(b.offset),
+      }
+    );
+  }
+
+  public get(lod: number, offset: Vector2) {
+    return this.store.get({ lod, offset });
+  }
+}
+
+interface TiledHeatmapVisProps {
+  api: TilesApi;
+  domain: Domain;
+  scaleType: ScaleType;
+  colorMap: ColorMap;
+  invertColorMap?: boolean;
+  displayLowerResolutions?: boolean;
+  qualityFactor?: number;
+}
+
+function TiledHeatmapVis(props: TiledHeatmapVisProps) {
+  const {
+    api,
+    domain,
+    scaleType,
+    colorMap,
+    invertColorMap = false,
+    displayLowerResolutions = true,
+    qualityFactor = 1,
+  } = props;
+  const { width, height } = useMemo(() => api.levelsSize[0], [api]);
+
+  return (
+    <VisCanvas
+      abscissaConfig={{
+        visDomain: [0, width],
+        isIndexAxis: true,
+        showGrid: false,
+      }}
+      ordinateConfig={{
+        visDomain: [0, height],
+        isIndexAxis: true,
+        showGrid: false,
+      }}
+      visRatio={1}
+    >
+      <PanMesh />
+      <ZoomMesh />
+      <TiledHeatmap
+        api={api}
+        domain={domain}
+        scaleType={scaleType}
+        colorMap={colorMap}
+        invertColorMap={invertColorMap}
+        displayLowerResolutions={displayLowerResolutions}
+        qualityFactor={qualityFactor}
+      />
+    </VisCanvas>
+  );
+}
+
+const Template: Story<TiledHeatmapVisProps> = (args) => {
+  return <TiledHeatmapVis {...args} />;
```

Comment:
You can just inline the `TiledHeatmapVis` component above in here. No need for the extra layer.

You can also import `TiledHeatmapProps` from the `TileHeatmap` component instead of duplicating the interface here.

Reply:
(You can keep destructuring the story args, though: `{...args}`)

Reply:
Great, that removes a lot of boiler plate!